### PR TITLE
Add support for authentication with Okta

### PR DESCRIPTION
**Pull request by jsatk**
I need support for Okta with Bell. I've currently forked it and have patched it in, but I think this would be useful for the mainline bell release.

Our use of Okta also depends on exposing the payload. It appears that that was already done in this PR. https://github.com/hapijs/bell/pull/273

I'd love to see https://github.com/hapijs/bell/pull/273 land and then this. 

I tried my best to follow all current style and best practices in Bell. Please @ldesplat let me know if you have any requests, feedback, etc.

### CODE REVIEW

File: lib/providers/index.js [hunk 1]
```diff
@@ -1,36 +1,37 @@
 'use strict';
 
 exports = module.exports = {
+    arcgisonline: require('./arcgisonline'),
```
Comment (AdriVanHoudt) [inline lib/providers/index.js]: This makes my brain smile 👌

**Comment by jsatk** [comment]
@AdriVanHoudt @ldesplat would love to see this merged or get the proper feedback to get this to a mergeable state so I can drop my private fork of this repo. Thanks!

**Comment by AdriVanHoudt** [comment]
This still needs an entry in https://github.com/hapijs/bell/blob/master/Providers.md

**Comment by jsatk** [comment]
Thanks for the feedback @AdriVanHoudt. Will add tomorrow!

**Comment by jsatk** [comment]
@AdriVanHoudt Just added documentation to https://github.com/hapijs/bell/blob/master/Providers.md,

Please let me know if there is anything else. Thanks!

File: lib/providers/okta.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    schema: Joi.object({
+        uri: Joi.string().uri().required()
+    }).required()
+};
+
+exports = module.exports = function (options) {
+
+    const results = Joi.validate(options, internals.schema);
+    Hoek.assert(!results.error, results.error);
+    const settings = results.value;
```
Comment (AdriVanHoudt) [inline lib/providers/okta.js]: Should you not assert there is an uri since you do not provide a default?

Comment (jsatk) [inline lib/providers/okta.js]: This is following a very similar pattern as https://github.com/hapijs/bell/blob/master/lib/providers/auth0.js#L12. 

It was my understanding that line 17 using `Joi.validate` would ensure folks pass in the expected schema.

**Comment by jsatk** [comment]
@AdriVanHoudt Awesome. Thanks so much for the review. I responded to your comment. Please let me know if there are any further questions.

And again, this PR requires https://github.com/hapijs/bell/pull/273 to go first and therefore is blocked by https://github.com/hapijs/bell/pull/273 merging.

File: lib/providers/okta.js (hunk 2 above)
Comment (AdriVanHoudt) [inline lib/providers/okta.js]: Oh I totally missed the `internals.schema` having uri in it 🙄

**Comment by AdriVanHoudt** [comment]
Well the dependency on #273 is just specific for your team no? I don't see a technical dependency?

**Comment by ldesplat** [comment]
Thanks for the PR @jsatk and for the review @AdriVanHoudt 

I'll make a release later this week with #273 as well.

**Comment by jsatk** [comment]
Thanks so much @ldesplat.

**Comment by jsatk** [comment]
@adrivanhoudt good point. You're correct. I'm thinking through the lense of what my team needs. 😂

**Comment by jsatk** [comment]
Hi @ldesplat & @AdriVanHoudt. Any update on when the 8.6.0 release will happen? Thanks!

**Comment by AdriVanHoudt** [comment]
@ldesplat said later this week so you'll have to wait until he has some time to do it, that is how OSS works ;)

**Comment by jsatk** [comment]
@AdriVanHoudt totally cool. Just asking. Thanks.

**Pull request merged**